django_project/base/views/project.py

```diff
@@ -289,3 +342,253 @@ def get_redirect_url(self, slug):
+    try:
+        dbrequest = Merchants.objects.values_list(
+            'merchantid', flat=True).get(
+            firstname=loggeduser)
+    except BaseException:
+        dbrequest = None
+
+    returnid = request.GET.get(
+        'code')
+    # Payout testing
+    #merchantid = "acct_1B4V9NJeuZPvjWZO"
+    #balance, b = merchantpayout(merchantid)
+    # platformpayout()
+    context = {
+        'ourclientid': ourclientid,
+        'balance': balance,
+        'b': b,
+    }
+    return render(request, 'payments/pay.html', context)
+
+
+@login_required
+def createaccount(request):
+    context = {
+
+    }
+    return render(request, 'payments/createaccount.html', context)
+
+
+@login_required
+def processcustomerandcharge(request):
+    firstname = request.user.username  # Look for logged in user
+    customerid = 0
+    merchantid = 0
+    try:
+        dbrequest = Customer.objects.values_list(
+            'merchantid', flat=True).get(
+            firstname=firstname)
+    except BaseException:
+        dbrequest = None
+
+    if dbrequest is None:  # The customer does not exist
+        # 1. Create user -> Click button
+        return redirect(createaccount)  # Create charge and account
+        # 2. Refresh processcustomerandcharge
+    else:
+        customerid = dbrequest
+        request.session['customerid'] = customerid
+        # Insert code to specify to whom the payment should go
+        # Get list of projects that have connected accounts
+
+        #1.Get list of usernames in merchant database.
+        firstnames = list(Merchants.objects.values_list('firstname',flat=True))
+        firstnamesid = list(User.objects.filter(username__in=firstnames).values_list('id',flat=True))
+        dbmerchreq = Project.objects.filter(owner_id__in=firstnamesid).values('id','name')
+        projects = Project.objects.exclude(owner_id__in=firstnamesid).values('id','name')
+        lengthout = projects.count
+        lengthin = dbmerchreq.count
+    context = {
+        'customerid': customerid,
+        'merchantid': merchantid,
+        'dbrequest': dbrequest,
+        'dbmerchreq': dbmerchreq,
+        'lengthin': lengthin,
+        'lengthout': lengthout,
+        'projects': projects
+    }
+    return render(request, 'payments/cust.html', context)
+
+
+@login_required
+def createmerchant(request):  # Call this method after Stripe OAuth
+    merchantid = 0
+    returnid = request.GET.get(
+        'code')  # token that is sent to retrieve client id to be stored in db
+    # POST request to OAUTH
+    print("returnid: %s"%returnid)
+    if returnid is not None:
+        r = requests.post(
+            "https://connect.stripe.com/oauth/token",
+            data={
+                'client_secret': stripe.api_key,
+                'code': returnid,
+                'grant_type': "authorization_code"})
+        temp = json.loads(r.content)
+        print(temp)
+        if r.status_code == 200:
+            print(r.status_code)
```

Comment:
see above comments about using print - rather use logging, or better, use a debugger to inspect variables at run time...